x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSort.java

```diff
@@ -219,8 +236,35 @@ public DataType dataType() {
 
     @Override
     public void validate(Failures failures) {
+        if (order == null) {
+            return;
+        }
         String operation = sourceText();
         failures.add(isFoldable(order, operation, SECOND));
+        if (isValidOrder() == false) {
+            failures.add(
+                Failure.fail(order, invalidOrderError, sourceText(), ASC.value(), DESC.value(), ((BytesRef) order.fold()).utf8ToString())
+            );
+        }
+    }
+
+    private boolean isValidOrder() {
```

Comment:
I don't think there is much value in having this separate method here. For one, `order.fold()` is likely called twice in case the order value is invalid.